src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -205,7 +205,24 @@ protected ChainedHeader FindHeaderToStartFrom(ChainedHeader fork)
             if ((chainedHeaderToStartFrom.Height - fork.Height) > MaxItemsPerHeadersMessage)
             {
                 // e.g. If fork = 3000 and tip is 6000 we need to start from block 5000.
-                var startFromHeight = fork.Height + MaxItemsPerHeadersMessage;
+                int startFromHeight = fork.Height + MaxItemsPerHeadersMessage;
+                chainedHeaderToStartFrom = chainedHeaderToStartFrom.GetAncestor(startFromHeight);
+            }
+
+            return chainedHeaderToStartFrom;
+        }
+
+        protected ChainedHeader FindHeaderToStartFromByHashStop(uint256 hashStop)
+        {
+            // Find the header
+            var hashStopHeader = this.consensusManager.Tip.FindAncestorOrSelf(hashStop);
+
+            // Do not return more than 2000 headers from the fork point.
+            ChainedHeader chainedHeaderToStartFrom = hashStopHeader;
+            if ((chainedHeaderToStartFrom.Height - hashStopHeader.Height) > MaxItemsPerHeadersMessage)
```

Comment:
This is odd are you not deducting the same value form itself?

Above you have `chainedHeaderToStartFrom = hashStopHeader`
Then under its `chainedHeaderToStartFrom.Height - hashStopHeader.Height`